network_listview_builder/lib/pokemonNotifier.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+import 'package:network_listview_builder/pokemonModel.dart';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+
+class PokemonNotifier extends ValueNotifier<List<Pokemon>> {
+  PokemonNotifier() : super(null);
+
+  int _pageNumber = 1;
+  bool _hasMorePokemon = true;
+  int _batchesOf = 5;
+  final String _apiUrl = 'https://pokeapi.co/api/v2/pokemon/';
+  List<Pokemon> _listPokemons;
+
+  @override
+  List<Pokemon> get value => _value;
+  List<Pokemon> _value;
+  @override
+  set value(List<Pokemon> newValue) {
+    _value = newValue;
+    notifyListeners();
+  }
+
+  Future<void> reload() async {
+    _listPokemons = <Pokemon>[];
+    _pageNumber = 1;
+    await httpGetPokemon(_pageNumber);
+  }
+
+  Future<void> getMore() async {
+    if (_hasMorePokemon) {
+      await httpGetPokemon(_pageNumber);
+    }
+  }
+
+  Future<void> httpGetPokemon(int page) async {
+    _listPokemons ??= <Pokemon>[];
+
+    for (var i = 0; i < _batchesOf; i++) {
+      http.Response res = await http.get(_apiUrl + (page + i).toString());
+
+      dynamic jsonDecoded = json.decode(res.body);
+
+      if (jsonDecoded != null) {
+        _listPokemons.add(Pokemon.fromJson(jsonDecoded));
+      } else {
+        _hasMorePokemon = false;
```

Comment:
instead of doing this, i think a while loop makes more sense in terms of readability.

`while(_hasMorePokemon) { .. }`

Should also inline this if statement for style points

Reply:
Umm dont think while works, this is just to check if we reach to a point where we have gone through all pokemons. And this check is to determine if it happens with in the batch of 5 calls. hasMorePokemon is not gonna turn false within all batches.

Reply:
I did this and it works

```
Future<void> httpGetPokemon(int page) async {
    _listPokemons ??= <Pokemon>[];
    int pageNumber = page;
    while (_hasMorePokemon && (pageNumber - page) < _batchesOf) {
      http.Response res = await http.get(_apiUrl + (pageNumber).toString());
      Map<String, dynamic> jsonDecoded = json.decode(res.body);
      jsonDecoded != null
          ? _listPokemons.add(Pokemon.fromJson(jsonDecoded))
          : _hasMorePokemon = false;
      pageNumber++;
    }

    _pageNumber = pageNumber;
    value = _listPokemons;
  }
```

Reply:
Also i just noticed, sometimes when you scroll the pokemon are duplicated, like ill see pikachu twice. Happened with your code, and still happening after I use this while loop. Might be some logic error in `reloadListView.dart`

Reply:
hmm is it, ill check